### PR TITLE
[bugfix] fix online dense export OOM in materialization and sanity check

### PR DESCRIPTION
**Pull request by eric-gecheng**
## Summary

Online dense export 在两个阶段会发生 host memory OOM：

**1. init_parameters 物料化阶段**

构建 CPU twin model 时，`init_parameters` 会在稀疏 lookup 被裁剪之前以完整 shape 物料化所有 meta 参数。dynamicemb / zch 表（1 亿+ 行）及 zch 尺寸的 MCH buffer 会耗尽 host memory，导致进程被 SIGKILL。

修复：在 `init_parameters` 之前调用 `_shrink_sparse_embedding_tables()`，将每个 EBC/EC 表权重替换为 1 行同 dtype Parameter，将 zch 尺寸的 MCH buffer 缩减为 1 行。warm-up batch 的稀疏 id 已清零，id 0 在 remap 后映射到 row 0，且稀疏 lookup 会在 graph surgery 后被裁剪，其值从不被消费。

**2. sanity check 的 pruned-graph forward 阶段**

`_run_dense_graph_sanity_check` 中的 `gm(sanity_data, device)` 调用使用代码生成的 `GraphModule.forward`，该方法的全部中间局部变量在函数返回前不会被释放。而同一函数中的 capture forward 使用 `torch.fx.Interpreter`（默认 `garbage_collect_values=True`），每个节点执行后立即释放不再引用的中间张量。两者峰值内存相差一个数量级以上，在训练模型已占用大量 host memory 的基础上导致 OOM。

修复：跳过 pruned-graph forward，仅保留 capture forward（已通过 Interpreter 验证完整图端到端正确性）。`torch.jit.script` 仍验证裁剪后图的结构正确性。

### CODE REVIEW

File: tzrec/utils/export_util.py [hunk 1]
```diff
@@ -1759,36 +1759,23 @@ def _run_dense_graph_sanity_check(
...
-        sanity_data[name] = values
-    for name, jt in seq_jts.items():
-        sanity_data[name] = jt.values().detach()
-        sanity_data[name + "__lengths"] = jt.lengths().detach()
-        batch_size = jt.lengths().size(0)
-    if batch_size:
-        sanity_data["batch_size"] = torch.tensor(batch_size, device=device)
-    with torch.no_grad():
-        gm(sanity_data, device)
+    # The pruned-graph forward (gm on captured sparse outputs) is skipped:
```
[the remark this answers is not in the document]
Comment (eric-gecheng) [reply tzrec/utils/export_util.py:1773]: Reject. It has been tested that torch.fx.Interpreter(gm).run(sanity_data, device) will still cause OOM.

**Review by tiankongdeguiji — APPROVED** [review]
